### ISSUE TITLE
Feature: Preview buttons

### ISSUE THREAD

**Issue by richard-kramer** [dominikks/discord-soundboard-bot#8]
It would be nice if I could locally play a sound to "preview" it without the need of switching the playback target. This would aid faster selection of the correct sound I want to play.

In my head this would simply be a "Play" button on every singe sound entry.

**Comment by dominikks** [comment]
In my mind, the sound entries are basically big play buttons already, so I'm not sure whether it makes sense to add an additional button within them. I think it might make it more difficult to hit the right sound fast, especially on mobile, as you'd need to make sure not to click the preview button by accident.

Could it be enough to make the switch between local and Discord playback easier by making it accessible via a button instead of the current dropdown? Additionally, it might make sense to add keybinds, like Shift+Click for local playback, or a hotkey to switch between local and Discord playback.

**Comment by richard-kramer** [comment]
`Shift+Click` sounds nice for desktop usage. But doesn't really address the problem on mobile. I'm actually not sure, if an easier switch would be sufficient and also, if an extra clickable surface on each button would hinder the "normal" click, if it's sized and positioned correctly. I guess, that's something that has to be tried out and tested. I'm not really a UX expert, but saw a similar design elsewhere.

**Comment by richard-kramer** [comment]
Another idea for mobile usage, if an extra button is out of the question: long pressing the sound buttons.

**Comment by dominikks** [comment]
Making a long-press work reliably on mobile across Chrome and Safari is unfortunately not as easy as you might think. 😅 I learned this the hard way recently.

I'll try to make it work with nested buttons first and see how well that works!

**Comment by richard-kramer** [comment]
Either nested buttons or a split interaction surface maybe? Like 3/4th of the button are for playing on the current playback target and 1/4th is the preview? Visually made distinct enough, so the user won't misclick? The ratios are examples of course. Maybe the width of the preview should also be fixed, so it has the same size on every button.

**Comment by richard-kramer** [comment]
I remembered, where I saw this before: Discord uses a similar Design and it feels pretty natural to use.

![image](https://github.com/dominikks/discord-soundboard-bot/assets/34233066/a64335ac-0081-460a-a45e-254c40772c6a)

The loudspeaker icon on the left is the "local" preview, while the rest of the button plays the sound for everyone. And this surface would be even bigger in this application, because there is no "favorite" button.
